lucas-marciano/vila-app
# feat: add loading state to button

# Introduction

This pull request introduces a loading state to the button component. 

## Changes

- Added a `LoadingButton` composable function that displays a loading indicator when clicked.
- Updated the UI to use the `LoadingButton` component.
- Added a state variable to control the loading state of the button.
- Updated the button's click handler to trigger the loading state and execute the desired action.

## Benefits

- Provides visual feedback to the user while the button's action is being performed.
- Improves the user experience by making it clear that the button is in a loading state.
- Prevents accidental clicks while the button is loading.

## How to test

1. Run the app.
2. Click the button "Acessar".
3. Observe that a loading indicator is displayed while the button's action is being performed.
4. Verify that the button returns to its normal state after the action is completed.

## Screenshots/GIFs

![Tela de Login](docs/gifs/login-screen.gif)
